client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java

```diff
@@ -260,4 +364,98 @@ private boolean retryConnect() {
+     * @param list java list
+     * @return nebula list
+     */
+    private NList list2Nlist(List<Object> list) {
+        NList nlist = new NList();
+        try {
+            for (Object item : list) {
+                nlist.values.add(value2Nvalue(item));
+            }
+        } catch (UnsupportedOperationException e) {
+            throw e;
+        }
+        return nlist;
+    }
+
+    /**
+     * convert java map to nebula thrift map
+     * @param map java map
+     * @return nebula map
+     */
+    private NMap map2Nmap(Map<String, Object> map) {
+        NMap nmap = new NMap();
+        try {
+            for (Map.Entry<String, Object> entry : map.entrySet()) {
+                nmap.kvs.put(entry.getKey().getBytes(), value2Nvalue(entry.getValue()));
+            }
+        } catch (UnsupportedOperationException e) {
+            throw e;
+        }
+        return nmap;
+    }
+
+
+    /**
+     * convert java value type to nebula thrift value type
+     * @param value java obj
+     * @return nebula value
+     */
+    private Value value2Nvalue(Object value) throws
+            UnsupportedOperationException {
+        Value nvalue = new Value();
+        if (value == null) {
+            nvalue.setNVal(NullType.__NULL__);
+        } else if (value instanceof Boolean) {
+            boolean bval = (Boolean)value;
+            nvalue.setBVal(bval);
+        } else if (value instanceof Integer) {
+            int ival = (Integer)value;
+            nvalue.setIVal(ival);
+        } else if (value instanceof Short) {
+            int ival = (Short)value;
+            nvalue.setIVal(ival);
+        } else if (value instanceof Byte) {
+            int ival = (Byte)value;
+            nvalue.setIVal(ival);
+        } else if (value instanceof Long) {
+            long ival = (Long)value;
+            nvalue.setIVal(ival);
+        } else if (value instanceof Float) {
+            float fval = (Float)value;
+            nvalue.setFVal(fval);
+        } else if (value instanceof Double) {
+            double dval = (Double)value;
+            nvalue.setFVal(dval);
+        } else if (value instanceof String) {
+            byte[] sval = ((String)value).getBytes();
+            nvalue.setSVal(sval);
+        } else if (value instanceof List) {
+            nvalue.setLVal(list2Nlist((List<Object>)value));
+        } else if (value instanceof Map) {
+            nvalue.setMVal(map2Nmap((Map<String, Object>)value));
+        } else if (value instanceof Value) {
+            return (Value)value;
+        } else if (value instanceof Date) {
+            return (Value)value;
+        } else if (value instanceof Time) {
+            return (Value)value;
+        } else if (value instanceof Duration) {
+            return (Value)value;
+        } else if (value instanceof DateTime) {
+            return (Value)value;
+        } else if (value instanceof Geography) {
+            return (Value)value;
```

Comment:
Date/Time/Duration/DateTime/Geography is not subclass of Value, cannot directly convert them to Value.

for Date type , use  `nvalue.setDtVal((Date)value)`

Reply:
ACK.